src/frontend/cypress/integration/placementOnCanvas.spec.ts

```diff
@@ -0,0 +1,154 @@
+        expectedElements.forEach(element => {
+            requestCanvasElementDomObject(element.id).should('exist');
+        });
+        let actualElements = checkAndGetCanvasElements([...expectedElements.values()]);
+        actualElements.then(actualElements => actualElements.forEach(actaulElement => {
+            let expectedElement = expectedElements.get(actaulElement.getId());
+            assert(actaulElement.getLeft() === expectedElement?.x, `x-coordinate match for ${expectedElement?.id}`);
+            assert(actaulElement.getTop() === expectedElement?.y, `y-coordinate match for ${expectedElement?.id}`);
+        }));
+    })
+})
+
+function createExpectedCanvasElements(data: string): Map<string, ExpectedCanvasElement> {
+    let result = new Map<string, ExpectedCanvasElement>();
+    data.split('\n').forEach(s => {
+        let newExpectedCanvasElement = new ExpectedCanvasElement(s);
+        result.set(newExpectedCanvasElement.id, newExpectedCanvasElement);
+    });
+    return result;
+}
+
+class ExpectedCanvasElement {
+    id: string;
+    x: number;
+    y: number;
+
+    constructor(csvLine: string) {
+        let fields: Array<string> = csvLine.split(',');
+        if(fields.length < 3) {
+            throw new Error(`Cannot create ExpectedCanvasElement because line has less than three fields: ${csvLine}`);
+        }
+        this.id = fields[0];
+        this.x = +fields[1];
+        this.y = +fields[2];
+        if(! Number.isInteger(this.x)) {
+            throw new Error(`ExpectedCanvasElement ${this.id} has invalid x-coordinate ${fields[1]}`);
+        }
+        if(! Number.isInteger(this.y)) {
+            throw new Error(`ExpectedCanvasElement ${this.id} has invalid y-coordinate ${fields[2]}`);
+        }
+    }
+
+    public toString(): string {
+        return `ExpectedCanvasElement ${this.id} at (${this.x}, ${this.y})`;
+    }
+}
+
+function checkAndGetCanvasElements(expected: Array<ExpectedCanvasElement>): Promise<Array<CanvasElement>> {
+    let promises: Array<Promise<CanvasElement>> = new Array<Promise<CanvasElement>>();
+    expected.forEach(item => promises.push(elementToCanvasElement(item.id)));
+    let result = Promise.all(promises);
+    result.then(items => {
+        cy.log('Have the actual canvas elements in CanvasElement objects:');
+        items.forEach(item => cy.log(item.toString()))
+    });
+    return result;
+}
+
+class CanvasElement {
+    constructor(
+        private id: string,
+        private left: number,
+        private top: number,
+        private width: number,
+        private height: number) {}
+
+    getId(): string {
+        return this.id;
+    }
+
+    getLeft(): number {
+        return this.left;
+    }
+
+    getTop(): number {
+        return this.top;
+    }
+
+    toString() {
+        return `CanvasElement(${this.left}, ${this.top}, ${this.width}, ${this.height})`;
+    }
+}
+
+function elementToCanvasElement(inputElementName: string): Promise<CanvasElement> {
```

Comment:
I read about async/await. I agree with you that this approach looks promising. I did not see how to use async/await here, because I am not working with ordinary promises but with Cypress chaining. Can we leave it like it is for now? Maybe I sort it out in another pull request.